pymrio/tools/iobalance.py

```diff
@@ -0,0 +1,189 @@
+        else:
+            raise MatrixNotCompatibleError("Final Demand shape is not compatible with InterIndustry Matrix")
+    else:
+        pass    
+        
+    
+    #Check Total Output
+    if isinstance(total_output, pd.DataFrame):
+        num_rows_X, num_cols_X = total_output.shape
+        if num_rows_X == num_rows_Z:
+            pass
+        else:
+            raise MatrixNotCompatibleError("Total Output shape is not compatible with InterIndustry Matrix")
+    else:
+        pass
+
+    print("Matrices Shapes are compatible")
+    
+    
+    #check if total Output is the sum of Industry and final demand outputs
+    for i in range(len(Inter_industry)):
+        
+        
+        Inter_industry_sum = Inter_industry.iloc[i].sum()
+        Final_demand_sum = final_demand.iloc[i].sum() 
+        total= Inter_industry_sum+Final_demand_sum
+        
+        
+        if round(total) == round(float(total_output.iloc[i])):
+            pass
+        else:
+            raise TotalOutputError('Total output does not equal sum of intermediate and final demand consumption')
+    
+    #check if system totals are balanced by summing total output and value added + InterIndustry
+    
+    X_sum = total_output.sum(axis=0) #Total Output sum
+    Z_sum_vertical = Inter_industry.sum(axis=0).sum() #Input Matrix sum (vertical sum of InterIndustry)
+    VA_sum = value_added.sum(axis=0).sum() #Value Added sum
+    
+    Input = Z_sum_vertical + VA_sum
+    Output = X_sum 
+    
+    #round to the nearest thousands decimal
+    if (Input/Output < 1.01).all() and (Input/Output > 0.99).all():
+        pass
+    else:
+        raise SystemNotBalancedError('The IO System is not Balanced: Sum of Inputs does not Equal sum of outputs')
+    
+    print('IO System is Balanced: Total Inputs equal Total Outputs')
+    
+    
+    #check if each sector in each region is balanced
+    for i in range(len(Inter_industry)):    
+        
+        #first check if the value added is multi-dimensional         
+        if value_added.ndim> 1:
+            #if yes then sum vertically and return a single dimensional array
+            value_added =value_added.sum(axis=0)
+        
+        else:
+            pass
+        
+        sector_Input_sum = float(Inter_industry.iloc[:, i].sum()) + float(value_added.iloc[i].sum(axis=0))
+        sector_output_sum = float(total_output.iloc[i])
+        
+        if int(sector_output_sum) == int(sector_Input_sum):
+            pass
+        
+        elif round(sector_output_sum) == round(sector_Input_sum):
+            pass
+        
+        #Here I allow for a 1% difference between sector Inputs and sector outputs
+        elif sector_Input_sum/sector_output_sum < 1.01 and sector_Input_sum/sector_output_sum > 0.99:
+            pass
+                
+        else:
+            print('Sector at column', i, 'is not balanced. The sum of the inputs is',sector_Input_sum,'While the sum of the outputs is',sector_output_sum ) 
```

Comment:
remove the print (and all others I might have missed). Just put the info the raise